src/System.IO.Pipelines/PipeOptions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    public class PipeOptions
```

Comment:
The only problem with making it a struct is that I'd have to make all of the properties set during construction. I don't really want to require that sine this was about making these parameters optional.